### PR TITLE
test: Increase test coverage

### PR DESCRIPTION
**Pull request by mgoldson**
# Description

- Adds a number of unit tests for core functionality.
- PR from before Node upgrade
- Rebased and updated to new functionality
- Removed tests and code that is no longer in use - See PR #118 
- Updates to data where dates needed fixing
- Updates to sinon stubs that weren't getting cleared down between test runs

Related issue: [RSP-1866](https://dvsa.atlassian.net/browse/RSP-1866)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -66,3 +66,5 @@ build/
 .vscode/
 
 cypress/videos
+
+.nyc_output
```
Comment (grjacob) [inline .gitignore:70]: Needs new line

File: .nycrc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "exclude": [
+        "src/public/",
+        "src/lambdas",
+        "test/**/*.spec.js",
+        "src/server/*js",
+        "src/server/validation",
+        "src/server/cookie-management"
+    ],
+    "reporter": [ "lcov", "text-summary" ]
+}
```
Comment (grjacob) [inline .nycrc:14]: Needs new line

**Comment by grjacob** [comment]
Does it need a version bump in the package.json?

**Review by jakedanielturner — APPROVED** [review]

File: test/data/parsedSinglePenalties.js [hunk 3]
```diff
@@ -2,9 +2,10 @@ export default [
...
     complete: true,
     reference: '0122451124578',
-    enabled: true,
+    enabled: false,
     paymentCode: '5ef305b89435c670',
     issueDate: '09/03/2018',
+    dateTime: 1520553600,
     vehicleReg: 'SD45TXT',
     formattedReference: '12245-1-124578-IM',
     location: 'Dalar Hir (A55 near Holyhead)',
```
Comment (ibrahimjassat) [inline test/data/parsedSinglePenalties.js:11]: Might want to review locations, as we might not have made it public knowledge where these sites are

**Review by ibrahimjassat — APPROVED** [review]

File: test/data/parsedSinglePenalties.js (hunk 3 above)
Comment (AnnaDodson) [reply test/data/parsedSinglePenalties.js:11]: The locations are already in the codebase. We can raise a seperate task to review test data.

**Pull request merged**